server/src/main/java/org/opensearch/common/settings/SettingsModule.java

```diff
@@ -179,6 +179,29 @@ public void configure(Binder binder) {
         binder.bind(IndexScopedSettings.class).toInstance(indexScopedSettings);
     }
 
+    /**
+     * Dynamically registers a new Setting at Runtime. This method is mostly used by plugins/extensions
+     * to register new settings at runtime. Settings can be of Node Scope or Index Scope.
+     * @param setting which is being registered in the cluster.
+     * @return boolean value is set to true when successfully registered, else returns false
+     */
+    public boolean registerDynamicSetting(Setting<?> setting) {
+        try {
+            registerSetting(setting);
```

Comment:
It seems like we now register settings in several places? What if `clusterSettings.registerSetting` or `indexScopedSettings.registerSetting` fails validation?

Reply:
> @reta Thats a good question. Ideally we should undo the changes we made with the call at L190. What do you think of moving this line to L197. This should be safe enough as the validation would throw an error before we register.

@saratvemulapalli this is an option, but in this case we won't register settings in `nodeSettings` and `indexSettings` since the previous statement just return. I was thinking of having the hooks, fe `onNodeSetting`, `onIndexSetting` (or more generic one), where the settings could be added after all validation checks.

Reply:
The settings functionality has been removed and added to a PR here: https://github.com/opensearch-project/OpenSearch/pull/5495